### PR TITLE
Feat: Implement image bundle on device level

### PR DESCRIPTION
**Pull request by colinmacgiolla**
## Change Summary

This PR allows for an image bundle to be attached/detached from a device

## Related Issue(s)

N/A

## Component(s) name

`arista.cvp.device_v3`

## Proposed changes
This change extends the existing device data model, to include an (optional) `imageBundle` key. 

If there is an existing image bundle (on device level) then the new `imageBundle` will replace the existing one, since there can only be a single image bundle - regardless of the `apply_mode`.

If the playbook is running in `apply_mode: strict`,  and no `imageBundle` is mapped in the vars, the device will have the  `imageBundle` removed.

## How to test
Tested on CVP 2021.2.2

Sample Playbook
```yaml
---
- name: Device Management in Cloudvision
  hosts: cv_server
  connection: local
  gather_facts: false
  collections:
    - arista.cvp
  vars:
    CVP_DEVICES:
      - serialNumber: XXXX
        parentContainerName: L2LEAF
        configlets:
            - 'AVD_Ipmi3'
        imageBundle: device_test_bundle
  tasks:
    - name: "Configure devices on {{inventory_hostname}}"
      arista.cvp.cv_device_v3:
        devices: '{{CVP_DEVICES}}'
        state: present
        search_key: serialNumber
        apply_mode: strict
```

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly. (check the box if not applicable)
- [ ] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Comment by sugetha24** [comment]
Can you address the failing CI tests? @colinmacgiolla

**Comment by colinmacgiolla** [comment]
How do I actually find the cause of this?
https://github.com/aristanetworks/ansible-cvp/runs/7866134885?check_suite_focus=true#step:6:220

**Comment by sugetha24** [comment]
Its here:
https://github.com/aristanetworks/ansible-cvp/runs/7866134885?check_suite_focus=true#step:6:188
i usually search for "error" or "fail" to find the cause

**Comment by colinmacgiolla** [comment]
Done

**Comment by colinmacgiolla** [comment]
Any more feedback on this one?

File: ansible_collections/arista/cvp/plugins/module_utils/resources/schemas/v3.py [hunk 1]
```diff
@@ -192,16 +192,13 @@
                     },
                     "imageBundle": {
                         "$id": "#/items/anyOf/0/properties/imageBundle",
-                        "type": "array",
-                        "title": "The imageBundle schema",
+                        "type": "string",
+                        "title": "The imageBundle name",
                         "description": "An explanation about the purpose of this instance.",
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/cvp/plugins/module_utils/resources/schemas/v3.py]: This looks like a generic description which should be updated or removed.

File: ansible_collections/arista/cvp/plugins/module_utils/resources/schemas/v3.py [hunk 2]
```diff
@@ -192,16 +192,13 @@
                     },
                     "imageBundle": {
                         "$id": "#/items/anyOf/0/properties/imageBundle",
-                        "type": "array",
-                        "title": "The imageBundle schema",
+                        "type": "string",
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/cvp/plugins/module_utils/resources/schemas/v3.py:193]: If this is already in use anywhere, this change would be breaking, correct?

Comment (ClausHolbechArista) [reply ansible_collections/arista/cvp/plugins/module_utils/resources/schemas/v3.py:193]: Also make sure to add this imageBundle key to any schema where the same data set might be exposed. Like we saw recently, the same data set is passed around so all the schemas has to take everything into account.

Comment (colinmacgiolla) [reply ansible_collections/arista/cvp/plugins/module_utils/resources/schemas/v3.py:193]: Correct - but it wasn't actually used anywhere before

File: ansible_collections/arista/cvp/plugins/module_utils/resources/schemas/v3.py (hunk 1 above)
Comment (colinmacgiolla) [reply ansible_collections/arista/cvp/plugins/module_utils/resources/schemas/v3.py]: yeah - updated now

File: ansible_collections/arista/cvp/plugins/module_utils/resources/schemas/v3.py (hunk 2 above)
Comment (sugetha24) [reply ansible_collections/arista/cvp/plugins/module_utils/resources/schemas/v3.py:193]: To colin's point, this was not exposed in the docs so maybe no one is using this unless they went looking for it in the code, but to Claus's point, this is a breaking change. @colinmacgiolla can we make this work without changing the type from `array` to `string` ?

Comment (colinmacgiolla) [reply ansible_collections/arista/cvp/plugins/module_utils/resources/schemas/v3.py:193]: Just to document here, in line with the discussions around #508 we will leave the type as `string`

**Review by sugetha24 — APPROVED** [review]
This LGTM! As mentioned in avd-developers GHC I don't think anyone is using the imageBundle keyword before so I'm ok with changing the schema.

**Review by sugetha24 — CHANGES_REQUESTED** [review]
Can you add a example playbook that tests this feature under the examples/ folder?

**Comment by sugetha24** [comment]
[Ready to merge: https://github.com/aristanetworks/ansible-cvp/pull/505](https://trello.com/c/XqSCn7vp/1-ready-to-merge-https-githubcom-aristanetworks-ansible-cvp-pull-505)

**Comment by colinmacgiolla** [comment]
> Can you add a example playbook that tests this feature under the examples/ folder?

Sure - updated that section with an example

**Comment by colinmacgiolla** [comment]
Any idea which file the markdown linter is complaining about now?

**Review by sugetha24 — APPROVED** [review]

File: ansible_collections/arista/cvp/examples/assign-image-bundle-to-device.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+---
```
Comment (sugetha24) [inline ansible_collections/arista/cvp/examples/assign-image-bundle-to-device.yaml]: Can you move these file into a folder `cv_image_v3` under the `examples\` and then add these examples?

File: ansible_collections/arista/cvp/docs/modules/cv_change_control_v3.rst.md [hunk 4]
```diff
@@ -74,7 +74,7 @@ The following options may be specified for this module:
 </table>
```
Comment (sugetha24) [inline ansible_collections/arista/cvp/docs/modules/cv_change_control_v3.rst.md]: Also, can you only add the .rst and .md files that has changes pertaining to the module you have changed(`cv_device_v3`) and remove the rest?

**Review by sugetha24 — APPROVED** [review]

**Review by noredistribution — APPROVED** [review]
LGTM!

**Pull request merged**